src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -85,3 +86,61 @@ export function euiNavLink(
     'data-test-subj': 'navDrawerAppsMenuLink',
   };
 }
+
+// Providing a buffer between the limit and the cut off index
+// protects from truncating just the last couple (6) characters
+const TRUNCATE_LIMIT: number = 64;
+const TRUNCATE_AT: number = 58;
+
+function truncateRecentItemLabel(label: string): string {
+  if (label.length > TRUNCATE_LIMIT) {
+    label = `${label.substring(0, TRUNCATE_AT)}…`;
+  }
+
+  return label;
+}
+
+/**
+ * @param {string} url - a relative or root relative url.  If a relative path is given then the
+ * absolute url returned will depend on the current page where this function is called from. For example
+ * if you are on page "http://www.mysite.com/shopping/kids" and you pass this function "adults", you would get
+ * back "http://www.mysite.com/shopping/adults".  If you passed this function a root relative path, or one that
+ * starts with a "/", for example "/account/cart", you would get back "http://www.mysite.com/account/cart".
+ * @return {string} the relative url transformed into an absolute url
+ */
+function relativeToAbsolute(url: string) {
+  const a = document.createElement('a');
+  a.setAttribute('href', url);
+  return a.href;
+}
+
+export type RecentNavLink = ReturnType<typeof createRecentNavLink>;
+
+export function createRecentNavLink(
+  recentLink: ChromeRecentlyAccessedHistoryItem,
+  navLinks: ChromeNavLink[],
+  basePath: HttpStart['basePath']
+) {
```

Comment:
Marking as resolved just because tests are still on the to-do list (at the top) but likely won't land until later this week (focusing on fixing functional issues first, then fixing existing tests, then writing new ones)